### ISSUE TITLE
PRIV_TOP vmod arguments are broken

### ISSUE THREAD

**Issue by nigoroll** [varnishcache/varnish-cache#3498]
When a `PRIV_TOP` argument is used for a vmod function like `$Function STRING test_priv_top(PRIV_TOP, STRING)`, we currently do not prevent it to be called from a wrong context, yet we still panic if the vcl contains such a call.

Introduced in 1851ae9acdbed813eaaf4729f92bcd3de53819da, shown by the test change below.

Note that also this does not conform to the documentation: https://varnish-cache.org/docs/6.0/reference/vmod.html#private-pointers

> When used from backend VCL subs, a NULL pointer will be passed.

```
diff --git a/bin/varnishtest/tests/v00043.vtc b/bin/varnishtest/tests/v00043.vtc
index 828b06668..6f985b037 100644
--- a/bin/varnishtest/tests/v00043.vtc
+++ b/bin/varnishtest/tests/v00043.vtc
@@ -81,7 +81,7 @@ varnish v1 -cliok "param.set debug +syncvsl" -vcl+backend {
 
        sub vcl_backend_response {
                set beresp.do_esi = true;
-
+               set beresp.http.x1 = debug.test_priv_top("");
        }
 
        sub vcl_deliver {
```

```
***  v1    debug|Error: Child (950) Panic at: Wed, 13 Jan 2021 16:45:07 GMT
***  v1    debug|Wrong turn at cache/cache_vrt_priv.c:243:
***  v1    debug|PRIV_TOP is only accessible in client VCL context
***  v1    debug|version = varnish-trunk revision 06a2e92e74764e05acd7fa77556755a1a24e952c, vrt api = 13.0
***  v1    debug|ident = Linux,4.9.0-14-amd64,x86_64,-jnone,-sdefault,-sdefault,-hcritbit,epoll
***  v1    debug|now = 30026.169678 (mono), 1610556307.524721 (real)
```

**Comment by nigoroll** [comment]
I have pondered how best to solve this:

If we just removed the `WRONG()` and returned `NULL` for `VRT_priv_top()` as before https://github.com/varnishcache/varnish-cache/commit/1851ae9acdbed813eaaf4729f92bcd3de53819da, we would have no useful user feedback at all. Also, the `WRONG()` is appropriate for programmatic use from vmods, which we should require to only call `VRT_priv_top()` from the right context.

So I am leaning towards vcc compile time checking and had a look how we could implement this: IMHO `vcc_priv_arg()` should use a facility similar to `vcc_AddCall()` / `vcc_AddUses()` to register that an argument has been used which requires client context, keeping in mind that for #3163 we do not only need compile time checking, but maintain the _okmask_ to mark context from which a call to the containing sub is allowed.

I lack VCC experience to see a good way how to approach this and would appreciate help.
